needlehaystack/providers/cohere.py

```diff
@@ -0,0 +1,57 @@
+import os
+import pkg_resources
+
+from operator import itemgetter
+from typing import Optional
+
+from cohere import Client, AsyncClient
+
+from .model import ModelProvider
+
+class Cohere(ModelProvider):
+    DEFAULT_MODEL_KWARGS: dict = dict(max_tokens          = 50,
+                                      temperature           = 0.3)
+
+    def __init__(self,
+                 model_name: str = "command-r",
+                 model_kwargs: dict = DEFAULT_MODEL_KWARGS):
+        """
+        :param model_name: The name of the model. Default is 'command-r'.
+        :param model_kwargs: Model configuration. Default is {max_tokens_to_sample: 300, temperature: 0}
+        """
+
+        api_key = os.getenv('NIAH_MODEL_API_KEY')
+        if (not api_key):
+            raise ValueError("NIAH_MODEL_API_KEY must be in env.")
+
+        self.model_name = model_name
+        self.model_kwargs = model_kwargs
+        self.api_key = api_key
+
+        self.client = AsyncClient(api_key=self.api_key)
+
+    async def evaluate_model(self, prompt: str) -> str:
+        response = await self.client.chat(message=prompt[-1]["message"], chat_history=prompt[:-1], model=self.model_name, **self.model_kwargs)
+        return response.text
+
+    def generate_prompt(self, context: str, retrieval_question: str) -> str | list[dict[str, str]]:
```

Comment:
would it be cleaner if this returned a touple of chat_history and message? that way you don't have to do the `-1` index acrobatics

Reply:
Good call, ty